src/api.rs

```diff
@@ -14,6 +17,42 @@ pub enum DriverType {
     Xen,
 }
 
+/// Supports passing initialization parameters to the driver
+///
+/// Some drivers can support optional extra initialization parameters.
+///
+/// This is required to initialize the KVM driver, which needs a `domain_name` and
+/// a `kvm_socket` parameters.
+///
+/// This is equivalent to LibVMI's `vmi_init_data_type_t`
+#[repr(C)]
+#[derive(Debug)]
+pub enum DriverInitParam {
+    KVMiSocket(String),
+}
+
+impl TryFrom<DriverInitParamFFI> for DriverInitParam {
+    type Error = IntoStringError;
+
+    fn try_from(param: DriverInitParamFFI) -> Result<Self, Self::Error> {
+        Ok(match param {
+            DriverInitParamFFI::KVMiSocket(cstr_socket) => DriverInitParam::KVMiSocket(
+                unsafe { CStr::from_ptr(cstr_socket) }
+                    .to_owned()
+                    .into_string()?,
+            ),
+        })
+    }
+}
+
+/// Support passing initialization options
+/// similar to DriverInitParam, however this enum offers C API compatibility
+#[repr(C)]
+#[derive(Debug)]
+pub enum DriverInitParamFFI {
```

Comment:
This should be part of the capi, not of the regular api imho.

Reply:
Fixed !